### PR TITLE
Table scan execution path code cleanup

### PR DESCRIPTION
**Pull request by xiaoxmeng**
No functional change and avoid RowSet copy passing around selective reader call stack

### CODE REVIEW

**Review by Yuhta — COMMENTED** [review]
Can you break down the change to multiple PRs?  GitHub starts to fold the change automatically and it's not practical to review now:

Screenshot 2024-09-05 at 10 08 33 AM

**Review by Yuhta — APPROVED** [review]
Reviewed internally with a button that can expand everything and show difference between the versions. :D

File: velox/dwio/common/ColumnSelector.cpp [hunk 1]
```diff
@@ -318,15 +315,15 @@ const FilterTypePtr& ColumnSelector::process(const std::string& column, bool) {
...
-    auto colName = name.substr(0, pos);
-    auto expr = name.substr(pos + 1);
-    return std::make_pair(colName, expr);
+  const auto pos = name.find('#');
+  if (pos == std::string::npos) {
+    return std::make_pair(name, "");
   }
 
-  return std::make_pair(name, "");
+  // special map column handling
```
Comment (zuyu) [inline velox/dwio/common/ColumnSelector.cpp:323]: nit: seems we keep consistent about the comment starting w/ `///`.

Comment (xiaoxmeng) [reply velox/dwio/common/ColumnSelector.cpp:323]: /// is for public comments in header file. // for private comment or implementation comments.

**Pull request closed without merging**